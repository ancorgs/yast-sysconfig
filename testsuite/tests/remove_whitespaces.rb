# encoding: utf-8

# Sysconfig module - testsuite
#
# remove_whitespaces tests
#
# testedfiles: Sysconfig.ycp
#
# $Id$
#
module Yast
  class RemoveWhitespacesClient < Client
    def main
      Yast.import "Testsuite"
      Yast.import "Pkg" # override packamanager
      Yast.import "Sysconfig"

      Testsuite.Test(lambda { Sysconfig.remove_whitespaces(nil) }, [{}, {}, {}], nil)
      Testsuite.Test(lambda { Sysconfig.remove_whitespaces("") }, [{}, {}, {}], nil)
      Testsuite.Test(lambda { Sysconfig.remove_whitespaces("    ") }, [
        {},
        {},
        {}
      ], nil)
      Testsuite.Test(lambda { Sysconfig.remove_whitespaces("  var") }, [
        {},
        {},
        {}
      ], nil)
      Testsuite.Test(lambda { Sysconfig.remove_whitespaces("  var  ") }, [
        {},
        {},
        {}
      ], nil)
      Testsuite.Test(lambda { Sysconfig.remove_whitespaces("v  a  r") }, [
        {},
        {},
        {}
      ], nil)
      Testsuite.Test(lambda { Sysconfig.remove_whitespaces("va r ") }, [
        {},
        {},
        {}
      ], nil)
      Testsuite.Test(lambda { Sysconfig.remove_whitespaces("var") }, [{}, {}, {}], nil)

      nil
    end
  end
end

Yast::RemoveWhitespacesClient.new.main
