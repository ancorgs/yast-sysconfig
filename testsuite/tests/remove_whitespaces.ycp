// Sysconfig module - testsuite
//
// remove_whitespaces tests
//
// $Id$
//

{

include "testsuite.ycp";

import "Sysconfig";

TEST( ``(Sysconfig::remove_whitespaces(nil) ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::remove_whitespaces("") ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::remove_whitespaces("    ") ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::remove_whitespaces("  var") ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::remove_whitespaces("  var  ") ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::remove_whitespaces("v  a  r") ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::remove_whitespaces("va r ") ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::remove_whitespaces("var") ), [ $[], $[], $[] ], nil );

}

