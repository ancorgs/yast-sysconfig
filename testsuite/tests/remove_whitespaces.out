Return	nil
Return	
Return	
Return	var
Return	var
Return	v  a  r
Return	va r
Return	var
